src/providers/openExchangeRates.ts

```diff
@@ -44,35 +45,44 @@ const openExchangeRates = async (
   }
 
   // Query
-  for (const date in datesAndCodesWanted) {
-    if (datesAndCodesWanted[date].length === 0) continue
-    const codes = datesAndCodesWanted[date].join(',')
-    const justDate = date.split('T')[0]
-    try {
-      const response = await fetch(
-        `${openExchangeRatesBaseUrl}/api/historical/${justDate}.json?app_id=${openExchangeRatesApiKey}&base=USD&symbols=${codes}`
-      )
-      const json = asOpenExchangeRatesResponse(await response.json()).rates
-      if (response.ok === false) {
-        log(
-          `openExchangeRates returned code ${response.status} for ${codes} at ${date}`
+  await Promise.all(
+    Object.keys(datesAndCodesWanted).map(async date => {
+      return new Promise<void>((resolve, reject) => {
+        if (datesAndCodesWanted[date].length === 0) resolve()
+        const codes = datesAndCodesWanted[date].join(',')
+        const justDate = date.split('T')[0]
+        resolve(
+          fetch(
+            `${openExchangeRatesBaseUrl}/api/historical/${justDate}.json?app_id=${openExchangeRatesApiKey}&base=USD&symbols=${codes}`
+          )
+            .then(async response => {
+              if (response.ok === false) {
+                log(
+                  `openExchangeRates returned code ${response.status} for ${codes} at ${date}`
+                )
+                throw new Error(
+                  `openExchangeRates returned with status: ${JSON.stringify(
+                    status ?? response.status
+                  )} and error: ${JSON.stringify(response.statusText)}`
+                )
+              }
+              const out = await response.json()
+              const json = asOpenExchangeRatesResponse(out).rates
+              console.log('70. json', JSON.stringify(json))
+              // Create return object
+              rates[date] = {}
+              for (const code of Object.keys(json)) {
+                rates[date][`${code}_USD`] = (1 / json[code]).toString()
+              }
+            })
+            .catch(e => {
+              log(`Failed to get ${codes} from openExchangeRates`, e)
+            })
         )
-        throw new Error(
-          `openExchangeRates returned with status: ${JSON.stringify(
-            status ?? response.status
-          )} and error: ${JSON.stringify(response.statusText)}`
-        )
-      }
+      })
+    })
+  )
```

Comment:
This is not how you should convert this into Promise.all.
Let's have a sidebar and I'll show you how it should be done.